internal/service/dynamodb/global_secondary_index_test.go

```diff
@@ -0,0 +1,1293 @@
+				ImportStateIdFunc:                    testAccGlobalSecondaryIndexImportStateFunc(resourceNameGSI),
+				ImportStateVerify:                    true,
+				ImportStateVerifyIdentifierAttribute: "index_name",
+			},
+		},
+	})
+}
+
+func TestAccDynamoDBGlobalSecondaryIndex_differentKeys(t *testing.T) {
+	ctx := acctest.Context(t)
+	var conf awstypes.TableDescription
+	var gsi awstypes.GlobalSecondaryIndexDescription
+
+	resourceNameTable := "aws_dynamodb_table.test"
+	resourceNameGSI := "aws_dynamodb_global_secondary_index.test"
+
+	rNameTable := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	rNameGSI := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	rHashKey := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	rRangeKey := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(ctx, t) },
+		ErrorCheck:               acctest.ErrorCheck(t, names.DynamoDBServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckTableDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlobalSecondaryIndexConfig_differentKeys(rNameTable, rNameGSI, rNameTable, ""),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckInitialTableExists(ctx, resourceNameTable, &conf),
+					testAccCheckInitialGSIExists(resourceNameGSI, &conf, &gsi),
+
+					resource.TestCheckResourceAttr(resourceNameGSI, "hash_key", rNameTable),
+					resource.TestCheckResourceAttr(resourceNameGSI, "hash_key_type", "S"),
+					func(s *terraform.State) error {
+						return resource.TestCheckResourceAttr(resourceNameGSI, names.AttrARN, aws.ToString(gsi.IndexArn))(s)
+					},
+					resource.TestCheckResourceAttr(resourceNameGSI, "index_name", rNameGSI),
+					resource.TestCheckNoResourceAttr(resourceNameGSI, "non_key_attributes"),
+					resource.TestCheckResourceAttr(resourceNameGSI, "projection_type", "ALL"),
+					resource.TestCheckNoResourceAttr(resourceNameGSI, "range_key"),
+					resource.TestCheckNoResourceAttr(resourceNameGSI, "range_key_type"),
+					resource.TestCheckResourceAttr(resourceNameGSI, "read_capacity", "1"),
+					resource.TestCheckResourceAttr(resourceNameGSI, "table", rNameTable),
+					resource.TestCheckResourceAttr(resourceNameGSI, "write_capacity", "1"),
+					resource.TestCheckResourceAttr(resourceNameGSI, "on_demand_throughput.#", "0"),
+					resource.TestCheckResourceAttr(resourceNameGSI, "warm_throughput.#", "0"),
+				),
+			},
+			{
+				ResourceName:      resourceNameTable,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"global_secondary_index",
+				},
+				ImportStateCheck: acctest.ComposeAggregateImportStateCheckFunc(
+					acctest.ImportCheckResourceAttr(names.AttrName, rNameTable),
+					acctest.ImportCheckResourceAttr("attribute.#", "1"),
+				),
+			},
```

Comment:
Since two back-to-back `ImportStateVerify` steps is an uncommon pattern, add a comment explaining what the step is for